src/datasources/accounts/accounts.datasource.ts

```diff
@@ -58,25 +94,47 @@ export class AccountsDatasource implements IAccountsDatasource {
     const { count } = await this
       .sql`DELETE FROM accounts WHERE address = ${address}`;
 
+    const { key } = CacheRouter.getAccountCacheDir(address);
+    await this.cacheService.deleteByKey(key);
+
     if (count === 0) {
       this.loggingService.debug(`Error deleting account ${address}: not found`);
     }
   }
 
+  getActiveDataTypes(): Promise<AccountDataType[]> {
+    const cacheDir = CacheRouter.getActiveAccountDataTypesCacheDir();
+    return this.getFromCacheOrExecuteAndCache<AccountDataType[]>(
+      cacheDir,
+      this.sql<
+        AccountDataType[]
+      >`SELECT * FROM account_data_types WHERE is_active IS TRUE`,
+      MAX_TTL,
+    );
```

Comment:
We have all data types in the cache so we could derive this from `getDataTypes`. Is there a benefit in caching the both?

Reply:
Yet, good point, maybe it's better to cache them all and just filter to get the active ones. Will do so 👌🏻

Reply:
I've removed the `AccountsDatasource.getActiveDataTypes`. In the end, it was just used internally, and as you suggested it's probably simpler to just store the data types and filter by activeness when needed. ([02d403d](https://github.com/safe-global/safe-client-gateway/pull/1760/commits/02d403db0a2a7c0ac65eacb6360270cbc6514a77))